src/hypothesis/internal/location.py

```diff
@@ -0,0 +1,37 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+from enum import Enum
+
+from hypothesis.control import _current_build_context
+
+
+class Location(Enum):
+    TOP_LEVEL = 0
+    INSIDE_DRAW = 1
+    INSIDE_TEST = 2
+
+
+def current_location():
```

Comment:
What else do you expect to use this for?  IMO the `s.example()` code would be as clean or cleaner with this logic inline, and YAGNI...